tensorflow_datasets/core/dataset_builder.py

```diff
@@ -365,13 +384,13 @@ def download_and_prepare(self, download_dir=None, download_config=None):
 
   @api_utils.disallow_positional_args
   def as_dataset(self,
-                 split=None,
-                 batch_size=None,
-                 shuffle_files=False,
-                 decoders=None,
-                 read_config=None,
-                 as_supervised=False,
-                 in_memory=None):
+                 split: Optional[splits_lib.SplitBase] = None,
+                 batch_size: Optional[int] = None,
+                 shuffle_files: Optional[bool] = False,
+                 decoders: Optional[Dict[str, decode.Decoder]] = None,
+                 read_config: Optional[read_config_lib.ReadConfig] = None,
+                 as_supervised: bool = False,
+                 in_memory: Optional[bool] = None) -> tf.data.Dataset:
```

Comment:
Return type should be `Nested[tf.data.Dataset]`

Reply:
Done